include/drivers/i2c.h

```diff
@@ -337,6 +337,151 @@ struct i2c_slave_config {
 	const struct i2c_slave_callbacks *callbacks;
 };
 
+#ifdef CONFIG_I2C_STATS
```

Comment:
should likely be included in Doxygen unconditionally, 
```suggestion
#if defined(CONFIG_I2C_STATS) || defined(__DOXYGEN__)
```